server/app/repository/VersionRepository.java

```diff
@@ -145,10 +145,7 @@ private Version publishNewSynchronizedVersion(PublishMode publishMode) {
           .forEach(
               program -> {
                 draft.addProgram(program);
-                if (settingsManifest.getProgramCacheEnabled()) {
-                  versionsByProgramCache.remove(String.valueOf(program.id));
-                  programCache.remove(String.valueOf(program.id));
-                }
+                removeCacheForProgram(String.valueOf(program.id));
```

Comment:
**suggestion (non-blocking):** instead of calling `String.valueOf` in several places, overload `removeCacheForProgram` so it accepts taking a long.

Reply:
Updated!